src/serial/parse_input.py

```diff
@@ -1,6 +1,35 @@
-def parse_string(message: str) -> str:
+import json
+
+ACC_GYR_Value = {'x': -1, 'y': -1, 'z': -1}
+LID_Value = {'left': -1, 'middle': -1, 'right': -1}
+COM_BAT_Value = {'x': -1}
+RPM_TMP_WET_GPS_Value = {'x': -1, 'y': -1}
+
+Arrays = [['x', 'y', 'z'], ['left', 'middle', 'right']]
 
-    split_message = message.split("-")
-    type = split_message.pop(0)
+def dataUpdater(value, myJson, splitMessage, array, size):
+    myData = value
+    i = 1
+    while i < size:
+        value = splitMessage[i]
+        myData[array[i - 1]] = value
+        i = i + 1
+    myJson["data"] = myData
+
+
+def parse_string(message: str) -> str:
+    if message is not None:
+        splitMessage = message.split("-")
+        myType = splitMessage[0]
+        myJson = {"type": myType, "data": None}
 
-    return {"type": type, "data": split_message}
+        if myType in ("ACC", "GYR"):
+            dataUpdater(ACC_GYR_Value, myJson, splitMessage, Arrays[0], len(splitMessage))
+        elif "LID" in splitMessage:
```

Comment:
is there a reason why we use `splitMessage` instead of how the other ones are? for readability it may be better just to use myType as well here

Reply:
👍🏻